src/github.com/couchbase/sync_gateway/db/changes.go

```diff
@@ -248,8 +250,25 @@ func (db *Database) MultiChangesFeed(chans base.Set, options ChangesOptions) (<-
 			// with access to both channels would see two versions on the feed.
 			for name, seqAddedAt := range channelsSince {
 				chanOpts := options
-				if seqAddedAt > 1 && options.Since.Before(SequenceID{Seq: seqAddedAt}) && options.Since.TriggeredBy == 0 {
-					// Newly added channel so send all of it to user:
+
+				// Check whether requires backfill based on addedChannels in this _changes feed
+				isNewChannel := false
+				if addedChannels != nil {
+					_, isNewChannel = (*addedChannels)[name]
+				}
+
+				// Check whether requires backfill based on current sequence, seqAddedAt
+				// Triggered by handling:
+				//   1. options.Since.TriggeredBy == seqAddedAt : We're in the middle of backfill for this channel, based
+				//    on the access grant in sequence options.Since.TriggeredBy.  Normally the entire backfill would be done in one
+				//    changes feed iteration, but this can be split over multiple iterations when 'limit' is used.
+				//   2. options.Since.TriggeredBy == 0 : Not currently doing a backfill
+				//   3. options.Since.TriggeredBy != 0 and <= seqAddedAt: We're in the middle of a backfill for another channel, but the backfill for
+				//     this channel is still pending.  Initiate the backfill for this channel - will be ordered below in the usual way (iterating over all channels)
+				requiresBackfill := seqAddedAt > 1 && options.Since.Before(SequenceID{Seq: seqAddedAt}) && (options.Since.TriggeredBy == 0 || options.Since.TriggeredBy < seqAddedAt)
```

Comment:
I think this line needs to be broken up either into multiple lines, or better yet maybe factor out some variables.  Currently there's too much cognitive load to mentally parse this.

For example:

```
backfillInProgress := options.Since.TriggeredBy != 0 && options.Since.TriggeredBy <= seqAddedAt
requiresBackfill := backfillInProgress && (!somethingElse || otherThing)
```